itests/gateway_test.go

```diff
@@ -558,3 +558,35 @@ func TestStatefulCallHandling(t *testing.T) {
 		req.ErrorContains(err, "filter not found")
 	}
 }
+
+func TestGatewayF3(t *testing.T) {
+	// Test that disabled & not-running F3 calls properly error
+
+	kit.QuietMiningLogs()
+
+	t.Run("F3 not running", func(t *testing.T) {
```

Comment:
Subjective: 
* Avoid including words that are inferred from the outer test name. In this case _I_ would drop `F3`
* Pick names such that when read in conjuction with outer test names they read like a user story (a poor man's lightweight behaviour driven test).
* Use `_` to separate context in which test being run to avoid test name conflict across tests in the same package, e.g. `TestGateway_F3`

Reply:
👍 changed inner test names, but opted to not use `_` because it's not the style of the current file or many of the existing itests